tests/unit/test_charm.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+from unittest import mock
+
+import ops
+import ops.testing
+import pytest
+from charms.operator_libs_linux.v2.snap import SnapError
+
+from charm import (
+    CLOUD_NAME,
+    OS_CLIENT_CONFIG,
+    SNAP_NAME,
+    OpenstackExporterOperatorCharm,
+)
+from service import SnapService
+
+
+class TestCharm:
+    def setup_method(self, _):
+        self.harness = ops.testing.Harness(OpenstackExporterOperatorCharm)
+
+    def teardown_method(self, _):
+        self.harness.cleanup()
+
+    @pytest.mark.parametrize(
+        "config",
+        [
+            ({"cache": False, "cache_ttl": "100s"}),
+            ({"cache": True, "cache_ttl": "200s"}),
+        ],
+    )
+    def test_config_changed(self, config, mocker):
```

Comment:
Not part of the formal review

I would consider writing documentation here for what the test is doing. You could use the `arrange/act/assert` framework for them